apps/mobile/src/hooks/useHandleCurrentGroup.js

```diff
@@ -10,19 +10,20 @@ import mixpanel from 'services/mixpanel'
 import useOpenURL from 'hooks/useOpenURL'
 import useConfirmAlert from 'hooks/useConfirmAlert'
 import useRouteParams from 'hooks/useRouteParams'
-import { modalScreenName } from 'hooks/useIsModalScreen'
 import getActiveRoute from 'navigation/getActiveRoute'
 
 export function useHandleCurrentGroupSlug () {
   const { currentGroupSlug } = useCurrentGroupStore()
   const { context, groupSlug, originalLinkingPath, pathMatcher } = useRouteParams()
   const [{ currentUser }] = useCurrentUser()
   const changeToGroup = useChangeToGroup()
-  const groupSlugFromPath = originalLinkingPath?.match(/\/groups\/([^\/]+)(?:\/|$)/)?.[1] ?? null
+  const pathMatches = originalLinkingPath?.match(/\/groups\/([^\/]+)(.*$)/)
+  const groupSlugFromPath = pathMatches?.[1] ?? null
+  const pathAfterMatch = pathMatches?.[2] ?? null
 
   useEffect(() => {
     if (currentUser && !currentGroupSlug && !groupSlugFromPath) {
-      changeToGroup(getLastViewedGroupSlug(currentUser))
+      changeToGroup(getLastViewedGroupSlug(currentUser)) // tempting to switch this to NoContextFallbackScreen
```

Comment:
what would this mean/do?

Reply:
- From cold-boot, the app will have no currentGroup set.
- It must open something
- So setting a group (in this case, the `lastViewedGroup`) effectively triggers side-effects to open the group at its home widget

There are a lot of 'depends on side-effects that aren't obvious" things happening for mobile navigation, a rich and bountiful terrain for bugs to thrive and persist. Some of those side-effects are coming from platform/framework behaviour and some from us having to layer them in based on how our app works

The NoContextFallback would instead not make the assumption that you want to see the last group but instead opens a (currently pretty ugly in full-screen) view of the global nav menu, but as a screen, not in the drawer. So it invites them to choose a context